### PR TITLE
examples: add zlib dependency

### LINKED ISSUE #972: 1.16.2: build fails because missing zlib.h
Looks like there is no anywhere in cmake files zlib detection despite fact that libheif examples are using zlib
<details>

```console
[ 56%] Building CXX object examples/CMakeFiles/heif-convert.dir/encoder_png.cc.o
cd /home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/x86_64-redhat-linux-gnu/examples && /usr/bin/g++ -DHAVE_JPEG_WRITE_ICC_PROFILE=1 -DHAVE_LIBJPEG=1 -DHAVE_LIBPNG=1 -DHAVE_LIBSHARPYUV=1 -DHAVE_UNISTD_H -DHAVE_VISIBILITY -DLIBHEIF_EXPORTS -DWITH_UNCOMPRESSED_CODEC=1 -I/home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/x86_64-redhat-linux-gnu -I/home/tkloczko/rpmbuild/BUILD/libheif-1.16.2 -I/home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/include/libheif -I/home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/include -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -DNDEBUG -std=c++11 -fPIE -fvisibility=hidden -fvisibility-inlines-hidden   -Wall -Werror -Wsign-compare -Wconversion -Wno-sign-conversion -Wno-error=conversion -Wno-error=unused-parameter -Wno-error=deprecated-declarations -MD -MT examples/CMakeFiles/heif-convert.dir/encoder_png.cc.o -MF CMakeFiles/heif-convert.dir/encoder_png.cc.o.d -o CMakeFiles/heif-convert.dir/encoder_png.cc.o -c /home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/examples/encoder_png.cc
/home/tkloczko/rpmbuild/BUILD/libheif-1.16.2/examples/encoder_png.cc:31:10: fatal error: zlib.h: No such file or directory
   31 | #include <zlib.h>
      |          ^~~~~~~~
compilation terminated.
```
</details>

### PR DESCRIPTION
**Pull request by bradh**
Resolves #972

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -119,8 +118,9 @@ if (PNG_FOUND)
             ../libheif/exif.cc
             common.cc
             common.h)
-    target_link_libraries(heif-thumbnailer heif ${PNG_LIBRARIES})
+    target_link_libraries(heif-thumbnailer heif ${PNG_LIBRARIES} ${ZLIB_LIBRARY})
```
Comment (kmilos) [inline examples/CMakeLists.txt:121]:
```suggestion
    target_link_libraries(heif-thumbnailer heif ${PNG_LIBRARIES} ${ZLIB_LIBRARIES})
```

**Review by kmilos — COMMENTED** [review]
Alternatively, maybe it's time to bump CMake minimum required version and simplifyby using imported targets ZLIB::ZLIB and PNG::PNG...

File: examples/CMakeLists.txt (hunk 1 above)
Comment (bradh) [reply examples/CMakeLists.txt:121]: Good catch. I copied-n-pasted from https://github.com/strukturag/libheif/blob/master/libheif/CMakeLists.txt#L117 

Will fix both.

**Comment by farindk** [comment]
> Alternatively, maybe it's time to bump CMake minimum required version and simplifyby using imported targets ZLIB::ZLIB and PNG::PNG...

Do you know what minimum cmake version is required for that? I think the current upper limit might be cmake v3.16.3 (Ubuntu 20.04).

**Comment by kmilos** [comment]
3.1 for ZLIB, 3.5 for PNG

**Comment by farindk** [comment]
The `#include <zlib.h>` in #972 is actually not needed. There is no dependency on zlib (except when header compression is turned on). I have removed the unnecessary include, this PR is not needed. I have still changed `ZLIB_LIBRARY` to `ZLIB_LIBRARIES`.

**Pull request closed without merging**